### PR TITLE
Setting up environment for Espresso based tests

### PR DESCRIPTION
**Pull request by yadav-rahul**
This PR includes:

1. Add `Espresso` dependency in `build.gradle`.
2. Test to check if animations are turned off in an Android device or not.

**Description**:
To avoid flakiness in Espresso tests, it is highly recommended that you turn off system animations on the virtual or physical device(s) used for testing.
On your device, under Settings->Developer options disable the following 3 settings:
- Window animation scale
- Transition animation scale
- Animator duration scale

More details about it could be found here https://google.github.io/android-testing-support-library/docs/espresso/setup/index.html.

### CODE REVIEW

**Comment by lognaturel** [comment]
Thanks! Could you please merge from master and re-run your build [here](https://circleci.com/gh/yadav-rahul/collect/39)? I don't think the failure is related to anything in this pull request but it would be good to double check. When you merge from master, you'll get the test summary copied in CircleCI so we can take a look if there are still failures.

@grzesiek2010 You've worked a bit with Espresso so could you please take a look at this?

**Comment by yadav-rahul** [comment]
OK @lognaturel. I am on it.

**Comment by grzesiek2010** [comment]
@lognaturel I'll review that once the problem with build is fixed.

**Comment by lognaturel** [comment]
@grzesiek2010 Could you please take a look? The only comment I have is that I think there should be a link to `https://github.com/google/iosched/blob/master/android/src/androidTest/java/com/google/samples/apps/iosched/DeviceStateTest.java` in the comments or if there was a narrative with some more context around it that would be even better.

File: collect_app/src/androidTest/java/org/odk/collect/android/DeviceStateTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    public ActivityTestRule<AboutPreferencesActivity> mActivityRule =
+            new ActivityTestRule<>(
+                    AboutPreferencesActivity.class);
+
+    @Test
+    public void animator_areDisabled() {
+        // Espresso tests can be flaky when animations are enabled. Test devices should have them
+        // disabled. This can be toggled on the Android Device by opening the Settings App:
+        // Enable Developer Options, then Developer Options -> Animator duration scale = Animation off
+        float animationSetting = -1;
```
Comment (grzesiek2010) [inline collect_app/src/androidTest/java/org/odk/collect/android/DeviceStateTest.java:32]: We don't need to initialize animationSetting here.

File: collect_app/src/androidTest/java/org/odk/collect/android/DeviceStateTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        assertFalse("Animations are enabled; they should be disabled when running tests, " +
+                animationSetting, animationSetting > 0);
+    }
+
+    @Test
+    public void windowAnimations_areDisabled() {
+        // Espresso tests can be flaky when animations are enabled. Test devices should have them
+        // disabled. This can be toggled on the Android Device by opening the Settings App:
+        // Enable Developer Options, then Developer Options -> Window animation scale = Animation off
+        float winAnimationSetting = -1;
```
Comment (grzesiek2010) [inline collect_app/src/androidTest/java/org/odk/collect/android/DeviceStateTest.java:61]: We don't need to initialize winAnimationSetting here.

File: collect_app/src/androidTest/java/org/odk/collect/android/DeviceStateTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        assertFalse("Animations are enabled; they should be disabled when running tests, " +
+                winAnimationSetting, winAnimationSetting > 0);
+    }
+
+    @Test
+    public void transitions_areDisabled() {
+        // Espresso tests can be flaky when animations are enabled. Test devices should have them
+        // disabled. This can be toggled on the Android Device by opening the Settings App:
+        // Enable Developer Options, then Developer Options -> Transition animation scale = Animation off
+        float transitionSetting = -1;
```
Comment (grzesiek2010) [inline collect_app/src/androidTest/java/org/odk/collect/android/DeviceStateTest.java:90]: We don't need to initialize transitionSetting here.

File: collect_app/src/androidTest/java/org/odk/collect/android/DeviceStateTest.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+ * Tests to ensure the device is setup properly to run the tests. Before making changes to a test
+ * or researching a failure, make sure this set of tests pass.
+ */
+@RunWith(AndroidJUnit4.class)
+public class DeviceStateTest {
+
+    @Rule
+    public ActivityTestRule<AboutPreferencesActivity> mActivityRule =
+            new ActivityTestRule<>(
+                    AboutPreferencesActivity.class);
```
Comment (grzesiek2010) [inline collect_app/src/androidTest/java/org/odk/collect/android/DeviceStateTest.java:25]: I think we can just use MainMenuActivity? Is tehere any reason why we should use AboutPreferencesActivity?

**Comment by lognaturel** [comment]
@yadav-rahul I'm going to close this for now but please feel free to reopen when you are ready to address @grzesiek2010's questions!

**Pull request closed without merging**